crates/cli/src/cli.rs

```diff
@@ -1,114 +1,110 @@
 mod api;
+mod license;
 mod resolver;
 mod utils;
 mod verify;
 
-use crate::api::LicenseType;
+use crate::license::LicenseType;
 use crate::utils::detect_local_tools;
-use crate::verify::VerifyProjectArgs;
 use camino::Utf8PathBuf;
-use clap::{Parser, Subcommand};
+use console::{style, Emoji};
 use dialoguer::{theme::ColorfulTheme, Confirm, Input, Select};
+use indicatif::{HumanDuration, ProgressStyle};
 use regex::Regex;
-use std::env;
+use std::{env, time::Instant};
 use strum::IntoEnumIterator;
+use verify::VerifyProjectArgs;
 
-#[derive(Parser)]
-#[command(author, version, about, long_about = None)]
-#[command(propagate_version = true)]
-struct Cli {
-    #[command(subcommand)]
-    command: Commands,
-}
-
-// TODO 1: There's a need of refactoring all this to reduce repetition.
-// TODO 2: support other types of project/file configurations
-//          - single file
-//          - non scarb
-//          - multiple contracts in project
-#[derive(Subcommand)]
-enum Commands {
-    #[command(about = "Builds the starknet-contract-verifier output")]
-    VerifyProject(VerifyProjectArgs),
-    // VerifyFile(VerifyFileArgs),
+#[allow(dead_code)]
+enum TargetType {
+    ScarbProject,
+    File,
 }
 
 fn main() -> anyhow::Result<()> {
-    // TODO: make this cli use a secure api
-    // let api_key = match env::var("API_KEY") {
-    //     Ok(api_key) => Some(api_key),
-    //     Err(_) => None,
-    // };
-
-    // let api_key = match api_key {
-    //     Some(key_values) => key_values,
-    //     None => {
-    //         println!("API_KEY not detected in environment variables. You can get one at https://forms.gle/34RE6d4aiiv16HoW6");
-    //         return Ok(());
-    //     }
-    // };
+    println!(
+        "{} {} Getting project information...",
+        style("[1/3]").bold().dim(),
+        Emoji("📝", "")
+    );
 
+    // Network selection
     let is_debug_network = env::var("DEBUG_NETWORK").is_ok();
-
     let network_items = if is_debug_network {
         vec!["Mainnet", "Sepolia", "Integration", "Local"]
     } else {
         vec!["Mainnet", "Sepolia"]
     };
-
-    let network_index: Option<usize> = Select::with_theme(&ColorfulTheme::default())
+    let network_index = Select::with_theme(&ColorfulTheme::default())
         .items(&network_items)
         .with_prompt("Which network would you like to verify on : ")
         .default(0)
-        .interact_opt()?;
+        .interact_opt()
+        .expect("Aborted at network selection, terminating...")
+        .expect("Aborted at network selection, terminating...");
```

Comment:
Since we want to separate the process into 2 parts, the resolving and the verifying, I think we can actually push the network selection after step 2, when resolving is successful.

I think it makes sense to only consider the network you want to verify on once the resolving actually is successful so they don't have to keep selecting network at startup? Wdyt?

Reply:
I agree, we can move network selection at the start of verify part